tests/bootstrap/test_cosmetic_configuration.py

```diff
@@ -164,7 +170,7 @@ def create_lambda_function(
             infra.add_custom_setup(
                 lambda: load_python_lambda_to_s3(
                     s3_client=aws_client.s3,
-                    bucket_name=infra.get_asset_bucket(),
```

Comment:
https://github.com/localstack/localstack/blob/b3686f173f6d7b5be2f7e4ffd3098c103c1cbce9/localstack/testing/scenario/provisioning.py#L102-L105

Reply:
I think this makes sense in this case, but we probably want to allow `infrastructure_setup` to take account/region. Then this helper would be OK I think? cc @dominikschubert 

See

https://github.com/localstack/localstack/blob/896ba4b3ff1f5ca6431dd5d9fb52c2b7f5a89d77/tests/bootstrap/conftest.py#L22-L37

We build a new client factory here without accepting region/account configuration.

Reply:
I'll make a note of it, since I'll soon try to rework the provisioner a bit for asset support as well :+1:  For now the solution seems fine, the bucket naming was mostly done for compatibility with live-AWS which isn't really relevant for bootstrap tests here